### PR TITLE
Update KDS example to use new connector api

### PR DESCRIPTION
**Pull request by guanlin84**
*Issue #, if available:*
N/A

*Description of changes:*
Update KDS example to use new connector api

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by hlteoh37 — APPROVED** [review]
LGTM - thanks @guanlin84  for the contribution!

**Comment by hlteoh37** [comment]
Let's not merge this PR until we run the internal E2E tests

File: java/KinesisConnectors/src/main/resources/flink-application-properties-dev.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[
+  {
+    "PropertyGroupId": "InputStream0",
+    "PropertyMap": {
+      "stream.arn": "arn:aws:kinesis:us-east-1:060795913058:stream/ExampleInputStream",
```
Comment (nicusX) [inline java/KinesisConnectors/src/main/resources/flink-application-properties-dev.json:5]: Please do not push real resource ARN. 
Use placeholders instead (not fictional ARN). This gives a more obvious failure when the user tries to run it locally without modifying the config

File: java/KinesisConnectors/src/main/resources/flink-application-properties-dev.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      "kinesis.stream.init.position": "LATEST",
+      "aws.region": "us-east-1"
+    }
+  },
+  {
+    "PropertyGroupId": "OutputStream0",
+    "PropertyMap": {
+      "stream.arn": "arn:aws:kinesis:us-east-1:060795913058:stream/ExampleOutputStream",
```
Comment (nicusX) [inline java/KinesisConnectors/src/main/resources/flink-application-properties-dev.json:13]: As above

**Review by nicusX — CHANGES_REQUESTED** [review]
Please do not hardwire ARN of real resources. Use placeholders instead

File: java/KinesisConnectors/src/main/resources/flink-application-properties-dev.json (hunk 2 above)
Comment (guanlin84) [reply java/KinesisConnectors/src/main/resources/flink-application-properties-dev.json]: Changed `stream.arn` values to use a dummy account id `012345678900`.

**Comment by nicusX** [comment]
@guanlin84 sorry for spotting these at a second iteration: 

Please, update the README too: `stream.name` was changed into `stream.arn`.
Also, while fixing this example, the README should also be fixed about `aws.region`. It erroneously says it's optional and it will fallback to the default region, but it is not correct. `aws.region` is (surprisingly) mandatory even when you specify the ARN

**Review by nicusX — APPROVED** [review]
Thanks for the contribution

**Pull request merged**